File: tests/test_cpg/fixtures/parity_ssrf.js
/** Cross-language SSRF fixture — JavaScript (Node.js). */

var http = require('http');

function proxy(req, res) {
    var url = req.query.url;  // $ source=ssrf
    http.get(url, function(resp) {  // $ sink=ssrf
        resp.pipe(res);
    });
}

function fetchUrl(req, res) {
    var target = req.query.target;  // $ source=ssrf
    var options = { hostname: target, path: '/api/data', method: 'GET' };
    var innerReq = http.request(options, function(innerResp) {  // $ sink=ssrf
        innerResp.pipe(res);
    });
    innerReq.end();
}

File: tests/test_cpg/fixtures/parity_ssrf.py
"""Cross-language SSRF fixture — Python."""

from flask import Flask, request

import requests

app = Flask(__name__)


@app.route("/proxy")
def proxy():
    url = request.args.get("url", "")  # $ source=ssrf
    resp = requests.get(url)  # $ sink=ssrf
    return resp.text


@app.route("/fetch")
def fetch():
    target = request.args.get("target", "")  # $ source=ssrf
    resp = requests.post(target, json={"data": "test"})  # $ sink=ssrf
    return resp.json()
